lib/validator.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+var HTTPError = require('./rbUtil').HTTPError;
+var ajv = require('ajv')();
+
+var inMapping = {
+    path: 'params',
+    query: 'query',
+    header: 'headers',
+    formData: 'body'
+};
+
+/**
+ * Converts a list of parameters from a swagger spec
+ * to JSON-schema for a request
+ *
+ * @param parameters list of params
+ * @returns JSON schema
+ */
+function convertToJsonSchema(parameters) {
+    var schema = {
+        type: 'object',
+        properties: {}
+    };
+
+    parameters.forEach(function(param) {
+        if (!schema.properties[inMapping[param.in]]) {
+            schema.properties[inMapping[param.in]] = {
+                type: 'object',
+                properties: {}
+            };
+            schema.required = schema.required || [];
```

Comment:
@gwicke no, according to the son-schema spec, required array must have at least on element. If we preinitialise it, it wouldn't compile if there's no required fields.